cli/openapi3_test.go

```diff
@@ -0,0 +1,498 @@
+				Description: "peach resource input",
+				Content: openapi3.Content{
+					"application/json": &openapi3.MediaType{
+						Schema: &openapi3.SchemaRef{Value: &openapi3.Schema{
+							Type: "object",
+							Properties: map[string]*openapi3.SchemaRef{
+								"color": {Value: &openapi3.Schema{
+									Description: "Peach color description",
+									Type:        "string",
+									ExtensionProps: openapi3.ExtensionProps{Extensions: map[string]interface{}{
+										"title": "Peach color title",
+									}},
+								}},
+							},
+							AdditionalPropertiesAllowed: boolean(false),
+						}},
+					},
+				},
+			}}
+
+			response := root.Paths["/peaches"].Post.RequestBody
+			Expect(response).To(WithTransform(transformJSON, Equal(transformJSON(expectedResponse))))
+		})
+
+		It("PUT single", func() {
+			expectedResponse := &openapi3.RequestBodyRef{Value: &openapi3.RequestBody{
+				Required:    true,
+				Description: "peach resource input",
+				Content: openapi3.Content{
+					"application/json": &openapi3.MediaType{
+						Schema: &openapi3.SchemaRef{Value: &openapi3.Schema{
+							Type: "object",
+							Properties: map[string]*openapi3.SchemaRef{
+								"sweetness": {Value: &openapi3.Schema{
+									Description: "Peach sweetness description",
+									Type:        "string",
+									ExtensionProps: openapi3.ExtensionProps{Extensions: map[string]interface{}{
+										"title": "Peach sweetness title",
+									}},
+								}},
+							},
+							AdditionalPropertiesAllowed: boolean(false),
+						}},
+					},
+				},
+			}}
+
+			response := root.Paths["/peaches/{id}"].Put.RequestBody
+			Expect(response).To(WithTransform(transformJSON, Equal(transformJSON(expectedResponse))))
+		})
+	})
+
+	It("operations should contains tags based on resource_group", func() {
+		multiple := root.Paths["/shops"]
+		single := root.Paths["/shops/{id}"]
+		operations := []*openapi3.Operation{
+			multiple.Get,
+			multiple.Post,
+			single.Get,
+			single.Put,
+			single.Delete,
+		}
+
+		expectedTags := []string{"Resource Group"}
+		for _, operation := range operations {
+			Expect(operation.Tags).To(Equal(expectedTags))
+		}
+	})
+})
+
+func removeRef(i interface{}) {
+	removeRefRecursive(reflect.ValueOf(i))
+}
+
+func removeRefRecursive(value reflect.Value) {
+	if !value.IsValid() {
+		return
+	}
+	switch value.Kind() {
+	case reflect.Ptr:
+		removeRefRecursive(value.Elem())
+
+	case reflect.Struct:
+		for i := 0; i < value.NumField(); i += 1 {
+			if value.Type().Field(i).Name == "Ref" {
+				value.Field(i).SetString("")
+			}
+			removeRefRecursive(value.Field(i))
+		}
+	case reflect.Slice:
+		for i := 0; i < value.Len(); i += 1 {
+			removeRefRecursive(value.Index(i))
+		}
+	case reflect.Map:
+		for _, key := range value.MapKeys() {
+			removeRefRecursive(value.MapIndex(key))
+		}
+	}
+}
+
+func transformJSON(value interface{}) interface{} {
+	// wrapping in some other object, allows arbitrary object to be marshaled and unmarshaled
```

Comment:
We want to compare JSON representations of these objects because this is what actually matters (there are some internal fields that may differ). For some of these checks `MatchJSON` could be used instead of `Equal`, but there is no alternative for `ContainElement`.